_posts/2024-04-29-wait-you-guys-are-writing-unit-tests/wait-you-guys-are-writing-unit-tests.Rmd

```diff
@@ -0,0 +1,95 @@
+---
+title: "Wait, you guys are writing unit tests?"
+description: |
+  A reminder.
+author:
+  - name: Chloe Pou-Prom
+    url: {}
+date: 2024-04-29
+output:
+  distill::distill_article:
+    self_contained: false
+categories:
+    - post-miscellaneous
+draft: true
+preview: unit_tests.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+![](unit_tests.png)
+
+Confession: I rarely write unit tests. 😅
+
+Yes, I know. They're important and they help find critical failures. Just enter "Are unit tests worth it?" in your favorite search engine, and you'll end up with a mountain ⛰️ of blog posts, StackOverflow replies, Reddit comments, heated discussions, etc.
+
+Overwhelmingly, the consensus is that *yes, unit tests are worth it*.
+
+And yet... a Very Official Survey[^1] shows that only 55% of data scientists write unit tests.
+
+[^1]: I asked chatgpt to make up a number. Maybe I'll poll our Slack channel to make this number more/less accurate.
+
+That being said, I did recently run into an issue that was identified by a unit test!!
+
+## What happened?
+
+**The situation**: We recently updated Data Virtuality (DV).
+
+> Data Virtuality (DV) is our data integration platform. It consolidates data from various hospital source systems and allows us to access real-time data from one platform. Many of our deployed solutions rely on DV.
+
+Anyway, we updated DV and...the upgrade went well! Applications continued running as expected!
+
+*Except*...
+
+I was working on a project post-update. Before pushing my changes to Gitlab, I ran the unit tests in my project...
+
+***And saw that one of the tests failed!!***
+
+## What *really* happened?
```

Comment:
I don't love the header 'what *really* happened' - e.g., we haven't been tricked by reality so far in this post.

Reply:
Changed